### PR TITLE
Fixed library joining logic to account for orgs

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@camhashemi @leogrim 

`LibraryCommander.joinLibrary` was one of the functions we were supposed to update to include organization logic. Pretty simple change.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala [hunk 1]
```diff
@@ -1112,11 +1112,14 @@ class LibraryCommanderImpl @Inject() (
       (lib, allInvites, existingActiveMembership)
     }
 
-    if (lib.kind == LibraryKind.SYSTEM_MAIN || lib.kind == LibraryKind.SYSTEM_SECRET) {
+    val isSystemGeneratedLibrary = lib.kind == LibraryKind.SYSTEM_MAIN || lib.kind == LibraryKind.SYSTEM_SECRET
+    val userCanJoinLibraryWithoutInvite = canViewLibrary(Some(userId), lib, authToken) // uses a db session
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:1116]: Someone double-check my thought process here. "joining" defaults to `READ_ONLY` unless there is explicitly greater permissions. I think we need to add a new `OrganizationPermission` to deal with the case where org members can automatically add themselves as collaborators to org libraries.

**Comment by andrewconner** [comment]
Mergeable?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:1116]: We probably want the default to be joining org libraries as collaborators if you're part of the org. You're right that we may want to encode that as an explicit permissions though.

**Comment by ryanpbrewster** [comment]
Yes, let's do this.

**Pull request merged**